services/libs/integrations/src/integrations/github/processStream.ts

```diff
@@ -0,0 +1,844 @@
+  const issuesQuery = new IssuesQuery(data.repo, ctx.integration.token)
+  const result = await issuesQuery.getSinglePage(data.page)
+
+  // filter out activities without authors (such as bots)
+  result.data = result.data.filter((i) => (i as any).author?.login)
+
+  // handle next page
+  await handleNextPageStream(ctx, result)
+
+  for (const record1 of result.data) {
+    const member = await prepareMember(record1.author, ctx)
+
+    await ctx.publishData<GithubApiData>({
+      type: GithubActivityType.ISSUE_OPENED,
+      data: record1,
+      member,
+      repo: data.repo,
+    })
+
+    // now need to parse issue events
+    for (const record2 of record1.timelineItems.nodes) {
+      switch (record2.__typename) {
+        case GithubPullRequestEvents.CLOSE: {
+          if (record2?.actor?.login) {
+            const member = await prepareMember(record2.actor, ctx)
+            await ctx.publishData<GithubApiData>({
+              type: GithubActivityType.ISSUE_CLOSED,
+              data: record2,
+              relatedData: record1,
+              member,
+              repo: data.repo,
+            })
+          }
+          break
+        }
+        default:
+          ctx.log.warn(
+            `Unsupported issue event:  ${record2.__typename}. This event will not be parsed.`,
+          )
+      }
+    }
+  }
+
+  // add each issue as separate stream for comments inside
+  for (const issue of result.data) {
+    await ctx.publishStream<GithubBasicStream>(
+      `${GithubStreamType.ISSUE_COMMENTS}:${issue.number}:firstPage`,
+      {
+        repo: data.repo,
+        page: '',
+        issueNumber: issue.number,
+      },
+    )
+  }
+}
+
+const processIssueCommentsStream: ProcessStreamHandler = async (ctx) => {
+  const data = ctx.stream.data as GithubBasicStream
+  const issueNumber = data.issueNumber
+  const issueCommentsQuery = new IssueCommentsQuery(data.repo, issueNumber, ctx.integration.token)
+  const result = await issueCommentsQuery.getSinglePage(data.page)
+  result.data = result.data.filter((i) => (i as any).author?.login)
+
+  // handle next page
+  await handleNextPageStream(ctx, result)
+
+  // get additional member info for each comment
+  for (const record of result.data) {
+    const member = await prepareMember(record.author, ctx)
+
+    // publish data
+    await ctx.publishData<GithubApiData>({
+      type: GithubActivityType.ISSUE_COMMENT,
+      data: record,
+      member,
+      repo: data.repo,
+    })
+  }
+}
+
+const processDiscussionsStream: ProcessStreamHandler = async (ctx) => {
+  const data = ctx.stream.data as GithubBasicStream
+  const discussionsQuery = new DiscussionsQuery(data.repo, ctx.integration.token)
+  const result = await discussionsQuery.getSinglePage(data.page)
+
+  result.data = result.data.filter((i) => (i as any).author?.login)
+
+  // handle next page
+  await handleNextPageStream(ctx, result)
+
+  // get additional member info for each comment
+  for (const d of result.data) {
```

Comment:
NIT: let's call it `discussion` instead of `d`

Reply:
done